### PR TITLE
LINK-1719 | Use signup endpoint when creating a signup

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
- Use signup bulk create endpoint to create a signup without group when only 1 signup is added to create signup form
- Redirect user to signup completed page after creating a signup

## Closes
[LINK-1719](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1719)

[LINK-1719]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/domain/signupGroup/summaryPage/SummaryPage.tsx [hunk 1]
```diff
@@ -142,14 +163,26 @@ const SummaryPage: FC<SummaryPageProps> = ({ event, registration }) => {
...
 
-                createSignupGroup(values, {
-                  onError: (error) =>
-                    showServerErrors(
-                      { error: JSON.parse(error.message) },
-                      'signup'
-                    ),
-                  onSuccess: (id) => goToSignupGroupCompletedPage(id as string),
-                });
+                if (values.signups.length == 1) {
```
Comment (harriris-vincit) [inline src/domain/signupGroup/summaryPage/SummaryPage.tsx:166]:
```suggestion
                if (values.signups.length === 1) {
```

File: src/domain/signup/SignupCompletedPage.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const SignupCompletedPageWrapper: React.FC = () => {
+  const { event, isLoading, registration } = useEventAndRegistrationData();
+  const { isLoading: isLoadingSignup, signup } = useSignupData();
+
+  return (
+    <LoadingSpinner isLoading={isLoading || isLoadingSignup}>
+      {event && registration && signup ? (
+        <SignupCompletedPage
+          event={event}
+          inWaitingList={signup.attendee_status == ATTENDEE_STATUS.Waitlisted}
```
Comment (harriris-vincit) [inline src/domain/signup/SignupCompletedPage.tsx:18]:
```suggestion
          inWaitingList={signup.attendee_status === ATTENDEE_STATUS.Waitlisted}
```

Comment (jorilindell) [reply src/domain/signup/SignupCompletedPage.tsx:18]: Too much Python I guess. Fixed

File: src/domain/signup/utils.ts [hunk 3]
```diff
@@ -197,6 +219,33 @@ export const getUpdateSignupPayload = ({
...
+  formValues: SignupGroupFormFields;
+  registration: Registration;
+  reservationCode: string;
+}): CreateSignupsMutationInput => {
+  const { contactPerson, signups: signupsValues } = formValues;
+
+  const signups: SignupInput[] = signupsValues.map((signupData, index) => ({
+    ...getSignupPayload({
+      formValues,
+      responsibleForGroup: index === 0,
```
Comment (harriris-vincit) [inline src/domain/signup/utils.ts:236]: `responsible_for_group` is no longer used by the back-end / it doesn't exist there anymore

File: src/domain/signupGroup/summaryPage/SummaryPage.tsx (hunk 1 above)
Comment (jorilindell) [reply src/domain/signupGroup/summaryPage/SummaryPage.tsx]: Too much Python I guess. Fixed

File: src/domain/signup/SignupCompletedPage.tsx (hunk 2 above)
Comment (jorilindell) [reply src/domain/signup/SignupCompletedPage.tsx]: Fixed this one as well

File: src/domain/signupGroup/summaryPage/__tests__/SummaryPage.test.tsx [hunk 4]
```diff
@@ -47,25 +48,40 @@ beforeEach(() => {
...
-  },
+const signupValues = {
+  city: 'City',
+  dateOfBirth: formatDate(subYears(new Date(), 9)),
+  extraInfo: '',
+  firstName: 'First name',
+  id: null,
+  inWaitingList: false,
+  lastName: 'Last name',
+  responsibleForGroup: true,
```
Comment (harriris-vincit) [inline src/domain/signupGroup/summaryPage/__tests__/SummaryPage.test.tsx:66]: Same here: this doesn't exist in the back-end anymore

File: src/domain/signup/utils.ts (hunk 3 above)
Comment (jorilindell) [reply src/domain/signup/utils.ts]: Removed responsibleForGroup in contact person related PR but it still slipped here. Removed

**Review by harriris-vincit — APPROVED** [review]
Looks very good to me. Also tested that this works as intended

**Pull request merged**